### ISSUE TITLE
org-lint: Unkown source block language

### ISSUE THREAD

**Issue by edgar-vincent** [flycheck/flycheck#2144]
## Checklist

- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description

In org-mode buffers, source blocks for languages installed as external packages (in this case `nix-mode`)  yield the following error:
`Unknown source block language: nix`.
However, `org-lint` itself doesn't report the error when run manually.

## Steps to reproduce

1. Install `nix-mode` from MELPA.
2. In an org-mode buffer, insert the following source block:
```
#+begin_src nix
# empty
#+end_src nix
```
3. flycheck raises the following warning: Uknown source block language: 'nix'

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior

`flycheck` shouldn't raise an warning.

## System configuration

```
Syntax checkers for buffer nixos.org in org-mode:

First checker to run:

  org-lint
    - may enable:         yes
    - executable:         Found at /nix/store/7dbq4m9bgdw17c1sz36y70zgmjjijqwx-emacs-pgtk-with-packages-30.2/bin/emacs
    - Org-lint available: yes (Org 9.7.11)

Checkers that are compatible with this mode, but will not run until properly configured:

  proselint (automatically disabled) reset
    - may enable: no
    - executable: Not found

  textlint (automatically disabled) reset
    - may enable:         no
    - executable:         Not found
    - configuration file: Not found
    - textlint plugin:    @textlint/text

Flycheck Mode is enabled.  Use SPC u C-c ! x to enable disabled
checkers.
--------------------

Flycheck version: 35.0
Emacs version:    30.2
System:           x86_64-pc-linux-gnu
Window system:    pgtk



```
Flycheck version: 35.0 367551e

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes

When I run `org-lint` in the same buffer, no errors are displayed.

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by bbatsov** [comment]
Yeah, I think the ultimately the problem is that you're running `org-lint` in batch mode, without the user's config. Perhaps we should just run with the config, but this will likely slow down things a lot if we're spinning an external Emacs process. Alternatively we can think of some way to just run it in the current process, which will certainly be the fastest and most correct approach as far as configuration goes.
